4단계 - 로또(2등)

victor-jo opened the pull request:
4단계 - 로또(2등) 리뷰 요청드립니다!

victor-jo committed:
refactor(lotto, lottoNumber): 전달된 파라미터에 대한 예외 처리 클래스 변경

- 전달된 파라미터가 분명하지 않을 경우 IllegalArgumentException 을 던지도록 수정

before:
    throw new RuntimeException
after:
    throw new IllegalArgumentException

victor-jo committed:
refactor(lottoMatchCount): 외부에서 사용하지 않는 변수에 대한 접근제한자 변경

- matchCount 변수 접근제한자 private 으로 변경

before:
    public int matchCount;
after:
    private int matchCount;

victor-jo committed:
docs(README_LOTTO): 로또 기능목록 추가

- 로또에 로또 번호 하나가 일치하는지 알 수 있는 기능

victor-jo committed:
test(lotto): 로또 기능 테스트 추가

- 로또에 로또 번호 하나가 일치하는지 알 수 있는 기능에 대한 테스트케이스 추가

victor-jo committed:
refactor(lottoMatchType): 로또 매칭 타입 보너스볼 기능 추가

- 매칭 타입 보너스볼인 경우 승진 메시지 추가
- 매칭 타입 전달되는 대표 문자열 메시지 추가

victor-jo committed:
refactor(lottoStatic): 로또 당첨 통계 보너스볼 기능 추가

- 불필요한 메소드 제거
- 보너스 로또 번호 추가 입력
- 보너스 로또 번호 판단 추가

victor-jo committed:
style(lottoApp): 로또 스캐너 메시지 재사용을 위해 이름 변경

- scanOneNumber 로 재사용 가능한 메시지 이름으로 변경

before:
    scanPayment() 메시지에서
after:
    scanOneNumber() 으로 변경

victor-jo committed:
feat(lottoApp): 로또앱 보너스볼 기능 추가

victor-jo committed:
refactor(lottoApp): 보너스볼 입력시 예외 처리

- 보너스볼 입력시 당첨번호와 중복된 번호 입력되면 에러 처리

joojimin commented on docs/README_LOTTO.md:
```diff
@@ -4,6 +4,7 @@
+  - 로또에 전달된 로또 번호 하나가 일치하는지 알려줘야 한다
 - 로또 상점은 로또 판매금액을 알고 있고, 제출된 돈 만큼 로또를 발급해야 한다
 - 로또 번호 생성 전략은 두가지이다
   - 랜덤하게 생성하는 전략
```
요구사항 업데이트 좋습니다! 💯

joojimin commented on src/main/java/lotto/Lotto.java:
```diff
@@ -11,7 +11,7 @@ public class Lotto {
     private Lotto(List<LottoNumber> lottoNumbers) {
         if (lottoNumbers.size() != LOTTO_NUMBER_COUNT) {
-            throw new RuntimeException(LOTTO_SIZE_ERROR);
+            throw new IllegalArgumentException(LOTTO_SIZE_ERROR);
```
피드백 반영 💯

joojimin commented on src/main/java/lotto/LottoApp.java:
```diff
@@ -34,9 +35,19 @@ public void run() throws IOException {
+        print(BONUS_NUMBER);
+        LottoNumber bonusLottoNumber = LottoNumber.valueOf(scanOneNumber());
+
+        if (winLotto.contains(bonusLottoNumber)) {
```
해당 로직이 보너스 번호 검증 로직인 것 같은데요! 어플리케이션의 동작을 담당하고있는 LottoApp의 run 메서드에서 노출이 되는게 관심사가 조금 달라보이는 것 같아요! 어떻게 개선할 수 있을까요?

joojimin reviewed (changes requested):
안녕하세요~ 원준님!
소소한 코멘트 하나 드렸는데 
확인해보시고 리뷰재요청 부탁드립니다!

victor-jo committed:
feat(bonusLottoNumber): 보너스 로또번호 객체 추가

- 보너스 로또번호 원시값 포장
- 보너스 로또번호 생성시 다른 로또와 겹치지 않게 생성
- 보너스 로또번호 생성시 다른 로또와 겹치지 않게 생성하는 테스트케이스 추가
- 보너스 로또번호가 스스로 전달된 로또와 매칭되는지 판단하는 메시지 추가

victor-jo committed:
refactor(lottoApp): 보너스볼 입력시 예외 처리 리팩터링

- 보너스볼 객체 생성시 예외 검사
- 보너스볼 객체 생성 후 전달하도록 리팩터링

victor-jo commented:
피드백 감사드립니다!
보너스 로또번호 자체를 포장하여 메시지를 전달받을 수 있도록 하여
예외처리는 생성자에서 처리하고
매칭여부는 match 메시지로 처리하도록 수정하였습니다

joojimin reviewed (approved):
미션진행 고생많으셨습니다!!
다음미션도 화이팅입니다!!
